src/main/java/org/apache/sling/feature/cpconverter/handlers/AbstractConfigurationEntryHandler.java

```diff
@@ -33,7 +33,7 @@
     private boolean enforceConfigurationBelowConfigFolder;
 
     public AbstractConfigurationEntryHandler(@NotNull String extension) {
-        super("/jcr_root/(?:apps|libs)/.+/(?<foldername>config|install)(\\.(?<runmode>[^/]+))?/(?<pid>.*)\\." + extension);
+        super("/jcr_root/(?:apps|libs)/.+/(?<foldername>config|install)(\\.(?<runmode>[^/]+))?(.*)/(?<pid>[^\\/]*)\\." + extension + ("(?<dir>.dir(/\\.content\\.xml)?)?$"));
     }
 
     void setEnforceConfgurationBelowConfigFolder(boolean enforceConfigurationBelowConfigFolder) {
```

Comment:
with the additional if for 'dir' the method gets quite complex. i would suggest to refactor it to reduce overall complexity

Reply:
no objecting a refactoring but  I only added an exit condition for the dir case - This shouldn't be block merging this bugfix in and can hopefully wait until I'm back from PTO